plugins/oauth/girder_oauth/__init__.py

```diff
@@ -17,15 +17,28 @@ def checkOauthUser(event):
     if user.get('oauth'):
         if isinstance(user['oauth'], dict):
             # Handle a legacy format where only 1 provider (Google) was stored
-            prettyProviderNames = 'Google'
+            userProviders = ['google']
         else:
+            userProviders = [val['provider'] for val in user['oauth']]
+
+        # Silently skip non-installed providers
+        availableUserProviders = [
+            userProvider for userProvider in userProviders
+            if userProvider in providers.idMap
+        ]
+
+        if availableUserProviders:
             prettyProviderNames = ', '.join(
-                providers.idMap[val['provider']].getProviderName(external=True)
-                for val in user['oauth']
+                providers.idMap[userProvider].getProviderName(external=True)
+                for userProvider in availableUserProviders
             )
+            helpMessage = 'Please log in with %s, or use the password reset link.' % \
+                          prettyProviderNames
+        else:
+            helpMessage = 'Please use the password reset link.'
+
         raise ValidationException(
-            "You don't have a password. Please log in with %s, or use the "
-            'password reset link.' % prettyProviderNames)
+            "You don't have a password. %s" % helpMessage)
```

Comment:
I believe this should turn into a single-quote string.

Reply:
No, [`flake8-quotes`](https://github.com/zheller/flake8-quotes) checks this, and it insists that we use double quotes when a single quote is within the string (to minimize escaping) and single quotes otherwise. The `tox -e lint` will always complain now if we get quotes wrong.

Reply:
Got it, missed the contraction there. Carry on.

Reply:
I use [black](https://black.readthedocs.io/en/stable/) on other projects, but it's hard to adopt into an already large codebase. Something to consider for the next major of Girder.

Reply:
Black doesn't handle quotes quite in the way we might want. It defaults to requiring double quotes. The only other option is to turn off quote checking entirely.

For this and other types of checks, Flake8 (and plugins) is still useful in a project with Black.